### DISCUSSION TITLE
Unable to authenticate with Apollo storage while fetching

### DISCUSSION THREAD

**Discussion by lukebryant** [apollographql/apollo-server#5110] [Q&A] [answered]
I'm getting this:

```
Unable to authenticate with Apollo storage while fetching https://storage-secrets.api.apollographql.com/REDACTED
...
Ensure that the API key is configured properly and that a federated service has been pushed.  For details, see https://go.apollo.dev/g/resolve-access-denied.
``` 

...when I deploy my federation gateway service. I'm pretty sure that my API key is set up correctly, because if I deploy with serviceList specified I get this:

```
A local gateway configuration is overriding a managed federation configuration.  To use the managed configuration, do not specify a service list or supergraphSdl locally. 
Apollo usage reporting starting! See your graph at https://studio.apollographql.com/graph/REDACTED
```

Apollo usage reporting wouldn't work if APOLLO_KEY wasn't correct, right?

APOLLO_KEY and APOLLO_GRAPH_VARIANT are the same in both instances.

I'm pretty sure a federated service for this graph variant has been pushed as well.

**Comment by lukebryant** [comment, marked as answer]
I think I've found the issue. When I actually sent a query to the version with serviceList which had logged "Apollo usage reporting starting!", I got this error message:

```
Error sending report to Apollo servers: APOLLO_KEY_REDACTED
 is not a legal HTTP header value
```
Note there is a line break after the key.

On googling it I came across this:
https://github.com/node-fetch/node-fetch/issues/401#issuecomment-565891390

So it seems like the issue is that I have a newline at the end of my secret.

Edit: Confirmed that was the issue. It's working fine now.

**Reply by glasser** [reply]
Are you replacing the key in the error message with APOLLO_KEY when pasting here? Because I don't know why we would literally be sending the string APOLLO_KEY (with or without a newline) to Apollo's servers...

**Reply by lukebryant** [reply]
Yes, I am. Sorry, I should have made that clearer. Just edited it to remove ambiguity.
